### PR TITLE
HPCC-12195 Document how to read JSON files

### PR DESCRIPTION
**Pull request by JamesDeFabia**
Signed-off-by: Jim DeFabia jamesdefabia@lexisnexis.com
@afishbeck Please review

### CODE REVIEW

File: docs/ECLLanguageReference/ECLR_mods/Recrd-DATASET.xml [hunk 1]
```diff
@@ -819,6 +819,125 @@ END;
...
+
+            <entry>A string constant containing the full XPATH to the tag that
+            delimits the records in the <emphasis>file</emphasis>.</entry>
+          </row>
+
+          <row>
+            <entry><emphasis role="bold">NOROOT</emphasis></entry>
+
+            <entry>Specifies the <emphasis>file</emphasis> is a JSON file with
+            no file tags, only row tags.</entry>
```
Comment (afishbeck) [inline docs/ECLLanguageReference/ECLR_mods/Recrd-DATASET.xml]: perhaps "no root level markup, only a collection of objects"?

File: docs/ECLLanguageReference/ECLR_mods/Recrd-DATASET.xml [hunk 2]
```diff
@@ -819,6 +819,125 @@ END;
...
+
+            <entry>A string constant containing the encryption key used to
+            create the file.</entry>
+          </row>
+        </tbody>
+      </tgroup>
+    </informaltable>
+
+    <para>This form is used to read a JSON file. The
+    <emphasis>xpath</emphasis> parameter defines the record delimiter tag
```
Comment (afishbeck) [inline docs/ECLLanguageReference/ECLR_mods/Recrd-DATASET.xml]: Something like "defines the path used to locate records within the JSON content"?  Unlike XML where a simple tag is indicated and scanned for, with JSON we require an explicit path.

File: docs/ECLLanguageReference/ECLR_mods/Recrd-DATASET.xml [hunk 3]
```diff
@@ -819,6 +819,125 @@ END;
...
+    <para>The key to getting individual field values from the JSON lies in the
+    RECORD structure<indexterm>
+        <primary>RECORD structure</primary>
+      </indexterm> field definitions. If the field name exactly matches a
+    lower case JSON tag containing the data, then nothing special is required.
+    Otherwise, <emphasis>{xpath(xpathtag)} </emphasis>appended to the field
+    name (where the <emphasis>xpathtag</emphasis> is a string constant
+    containing standard XPATH syntax) is required to extract the data. An
+    XPATH consisting of empty quotes ('') indicates the field receives the
+    entire record. An absolute XPATH is used to access properties of parent
```
Comment (afishbeck) [inline docs/ECLLanguageReference/ECLR_mods/Recrd-DATASET.xml]: child rather than parent?

**Comment by JamesDeFabia** [comment]
@afishbeck Edits made, please have another look

**Comment by afishbeck** [comment]
@JamesDeFabia looks good

**Pull request merged**